app/models/manageiq/providers/amazon/inventory/collector/cloud_manager.rb

```diff
@@ -74,10 +76,9 @@ def extra_image_references
     # public_images
     return @extra_image_references if @extra_image_references
 
-    db_image_refs        = Set.new(manager.miq_templates.pluck(:ems_ref))
     instances_image_refs = Set.new(instances.map { |x| x["image_id"] })
     api_images_refs      = Set.new((private_images + shared_images + public_images).map { |x| x["image_id"] })
 
-    @extra_image_references = ((db_image_refs + instances_image_refs) - api_images_refs).to_a
+    @extra_image_references = (instances_image_refs - api_images_refs).to_a
```

Comment:
So will this never pull in public images which are not in use by one of the instances?  Does that match with the use-case of provisioning from a public image?

Reply:
we will still pull in public images, if those are allowed, byt the specified filter

This is if the public images were not collected even then (either public images are disabled, or filter misses them). So we do have an old BZ saying that we always want image, because without image, we won't see OS info). 

So without this code, if you delete all Vms using the public image, we will disconnect it. Previously it would stay, but that is probably not possible to make in an efficient way.

Reply:
Ahh okay I misunderstood, :+1: thanks